addons/sale_project/models/sale_order_line.py

```diff
@@ -236,19 +236,27 @@ def _timesheet_create_task_prepare_values(self, project):
             allocated_hours = self._convert_qty_company_hours(self.company_id)
         sale_line_name_parts = self.name.split('\n')
         title = sale_line_name_parts[0] or self.product_id.name
-        description = '<br/>'.join(sale_line_name_parts[1:])
-        return {
+        res = {
             'name': title if project.sale_line_id else '%s - %s' % (self.order_id.name or '', title),
             'analytic_account_id': project.analytic_account_id.id,
             'allocated_hours': allocated_hours,
             'partner_id': self.order_id.partner_id.id,
-            'description': description,
             'project_id': project.id,
             'sale_line_id': self.id,
             'sale_order_id': self.order_id.id,
             'company_id': project.company_id.id,
             'user_ids': False,  # force non assigned task, as created as sudo()
         }
+        if 'calendar_event_id' in self\
+                and 'planned_date_begin' in self.env["project.task"]\
+                and 'date_deadline' in self.env["project.task"]\
+                and self.calendar_event_id.start\
+                and self.calendar_event_id.stop:
```

Comment:
1) I don't get the 3 first conditions? Is it related to access checking?
2) About the formatting, here's what I suggest. It is more readable IMO, but you're free to disagree :slightly_smiling_face: :
```suggestion
        if (
            'calendar_event_id' in self
            and 'planned_date_begin' in self.env["project.task"]
            and 'date_deadline' in self.env["project.task"]
            and self.calendar_event_id.start
            and self.calendar_event_id.stop
        ):
```